kernel/src/hil/crc.rs

```diff
@@ -0,0 +1,35 @@
+// Generic interface for CRC computation
+
+use returncode::ReturnCode;
+
+#[derive(Copy, Clone)]
+pub enum Polynomial {
+	CCIT8023,   // Polynomial 0x04C11DB7
+	CASTAGNOLI, // Polynomial 0x1EDC6F41
+	CCIT16,     // Polynomial 0x1021
+}
+
+pub fn poly_from_int(i: usize) -> Option<Polynomial> {
+    match i {
+        0 => Some(Polynomial::CCIT8023),
+        1 => Some(Polynomial::CASTAGNOLI),
+        2 => Some(Polynomial::CCIT16),
+        _ => None
+    }
+}
+
+pub trait CRC {
+    // Get the version of the CRC unit
+    fn get_version(&self) -> u32;
```

Comment:
`get_version` is pretty specific to a particular hardware implementation. Conversely, maybe many chips have a version, and we just don't promise anything about the meaning of the value (and therefore those that don't have a version can just return some dummy value).